fixtures/commas_in_pattern.json

```diff
@@ -1,22 +1,22 @@
-{ 
+{
+  "$schema": "http://json-schema.org/draft/2020-12/schema",
   "$id": "https://github.com/invopop/jsonschema/pattern-test",
   "$ref": "#/$defs/PatternTest",
-  "$schema": "http://json-schema.org/draft/2020-12/schema",
   "$defs": {
     "PatternTest": {
-      "required": [
-        "with_pattern"
-      ],
       "properties": {
         "with_pattern": {
+          "type": "string",
           "maxLength": 50,
           "minLength": 1,
-          "pattern": "[0-9]{1,4}",
-          "type": "string"
+          "pattern": "[0-9]{1,4}"
```

Comment:
I'm not making sense of this spec.

Is `pattern` anchored to the beginning/end of the string?

If so, isn't the `maxLength` effectively 4?

And if not, isn't this the same as `"pattern": "[0-9]"`?

That is to say:  If the pattern isn't anchored, then in effect, we're just looking for a single digit in the string?  And if it is anchored, then we'll never have a lenght beyond 4...

But maybe I'm misunderstanding.

Reply:
That's correct, but this is just a test for the examples, the content isn't that meaningful :-)